server/graphql/v2/input/OrderUpdateInput.ts

```diff
@@ -0,0 +1,27 @@
+import { GraphQLInputObjectType, GraphQLInt, GraphQLString } from 'graphql';
+
+export const OrderUpdateInput = new GraphQLInputObjectType({
+  name: 'OrderUpdateInput',
+  fields: () => ({
+    id: {
+      type: GraphQLString,
+      description: 'The public id identifying the order (ie: dgm9bnk8-0437xqry-ejpvzeol-jdayw5re)',
+    },
+    legacyId: {
+      type: GraphQLInt,
+      description: 'The legacy public id identifying the order (ie: 4242)',
+    },
+    amount: {
+      type: GraphQLInt,
```

Comment:
No!

Reply:
Can we have `AmountInput` back please.

Reply:
We can. But we'll also add validation to ensure the currency matches the order.
Also, keep in mind we only use `AmountInput` in Order, so this is consistent only within this object but inconsistent with the rest of the inputs.

Reply:
If we're not using `Amount` / `AmountInput` elsewhere, that's a problem.

We created this pattern so we stop having to think if an amount is in cents or in plain value, and to stop wondering in which currency it's expressed.

Any new development should use this pattern.